subsys/net/lib/coap/coap_server.c

```diff
@@ -170,6 +169,43 @@ static int coap_server_process(int sock_fd)
 
 	type = coap_header_get_type(&request);
 
+	if (received > sizeof(buf)) {
+		/* The message was truncated and can't be processed further */
+		uint8_t resp_buf[CONFIG_COAP_SERVER_MESSAGE_SIZE];
```

Comment:
Could we re-use the already allocated `buf`? We are using block transfer with payload size of 1024 bytes in our application, so this would allocate another 1100 bytes on the stack (message size includes the header and must be larger than the maximum expected payload size).

Reply:
Updated to re-use the `buf` array.